lib/foreman/cron.rb

```diff
@@ -0,0 +1,83 @@
+require 'rake'
+
+module Foreman
+  module Cron
+    TASKS_BY_CADENCE = {
+      hourly:  [],
+      daily:   %w[
+        reports:daily
+        db:sessions:clear
+        reports:expire
+        audits:expire
+      ],
+      weekly:  %w[
+        reports:weekly
+        notifications:clean
+      ],
+      monthly: %w[
+        reports:monthly
+      ],
+    }.freeze
+
+    class << self
+      def register(cadence, task_name)
+        cadence = cadence.to_sym
+
+        unless TASKS_BY_CADENCE.key?(cadence)
+          logger.warn(
+            "Foreman::Cron[#{cadence}]: unknown cadence when registering #{task_name}, ignoring"
+          )
+          return
+        end
+
+        extra_tasks[cadence] << task_name
```

Comment:
This would in theory allow registering a single task multiple times. Is that intentional?